daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -127,6 +138,7 @@ func ValidateLogOpt(cfg map[string]string) error {
 		switch key {
 		case "max-file":
 		case "max-size":
+		case "compression":
```

Comment:
It seems like we need some extra validation on this option.
Namely that it doesn't make sense if max-size is not set or if max-file is 0 or 1.

Reply:
Will do this, but I think we should valid this in 
 `func New(info logger.Info) (logger.Logger, error) `, `ValidateLogOpt` seems to valid the option name only.